### ISSUE TITLE
[Bug]: Audience absence should not trigger Error 500 

### ISSUE THREAD

**Issue by dreznichuk** [oauth2-proxy/oauth2-proxy#2808]
### OAuth2-Proxy Version

v7.7.0

### Provider

keycloak-oidc

### Expected Behaviour

When user token have no audience at all or have no valid audience, `403 Forbidden` should be returned

### Current Behaviour

`500 Internal Server error` is returned with log like this:
```
[oauthproxy.go:889] Error creating session during OAuth2 callback: audience from claim aud with value [test] does not match with any of allowed audiences map[oauth2-proxy:{}]
```

or if i have no audiences assigned:
```
[oauthproxy.go:830] Error creating session during OAuth2 callback: audience claims [aud] do not exist in claims: 
```

### Steps To Reproduce

1. typical oauth2proxy setup
2. typical Keycloak client, named oauth2-proxy
3. Disable `Full scope allowed` for oauth2-proxy keycloak client
4. Create Client scope with audience `oauth2-proxy` mapper
5. Assign role to this scope
6. Do **not** assign this role to any user
7. Add created scope as default to oauth2-proxy client
8. Try to login

### Possible Solutions

As I understand it, the audience field specifies which application the token is intended for. Therefore, if I receive a token that is not meant to be used with this application, meaning it does not contain the required audience, I would expect to see a "Permission denied" error rather than a server error.

**Comment by JoelSpeed** [comment]
Agreed, these errors should be surface as 403s rather than 500s, is anyone able to raise a PR to resolve the issue?
